salt/edx/edx_mysql.sls

```diff
@@ -0,0 +1,30 @@
+#!jinja|yaml
+
+{% from "orchestrate/aws_env_macro.jinja" import ENVIRONMENT with context %}
+{% set environment = salt.pillar.get('environments:{}'.format(ENVIRONMENT)) %}
+{% set purposes = environment.purposes %}
+{% set edxlocal_databases = {
+  'XQUEUE_MYSQL_DB_NAME': 'xqueue',
+  'EDXAPP_MYSQL_DB_NAME': 'edxapp',
+  }
+%}
+
+{% set edxapp_mysql_host = 'mysql.service.{}.consul'.format(environment) %}
+{% set edxapp_mysql_port = 3306 %}
+
+{% for db,name in edxlocal_databases.iteritems() %}
+{% for purpose in purposes %}
+{% set edxapp_mysql_creds = salt.vault.read(
+    'mysql-{env}/creds/edxapp-{purpose}'.format(
+        env=environment,
+        purpose=purpose)) %}
+edxapp_create_db_{{ name }}_{{ purpose }}:
+  mysql_database.present:
+    - name: {{ name }}_{{ purpose }}
```

Comment:
You will need to do a replace in the purpose string as well. `|replace('-', '_')`